install_script.sh.template

```diff
@@ -378,6 +378,214 @@ function ensure_config_file_exists() {
+            "distribution": "${distribution}",
+            "architecture": "$(uname -m)",
+            "kernel_version": "$(uname -v)",
+            "kernel_name": "$(uname -s)",
+            "kernel_release": "$(uname -r)"
+        },
+        "application": {
+            "service_name": "datadog-linux-install-script",
+            "service_version": "${install_script_version}",
+            "language_name": "UNKNOWN",
+            "language_version": "n/a",
+            "tracer_version": "n/a"
+        },
+        "payload": {
+            "logs": ${json_logs}
+        }
+    }
+END
+)
+
+    if ! echo "$telemetry_logs" | curl -f -sSL --retry 5 -o /dev/null \
+        "$telemetry_url" \
+        --header 'Content-Type: application/json' \
+        --header "DD-Api-Key: $apikey" \
+        --data @-
+    then
+        echo "Unable to send logs telemetry\n"
+    fi
+
+    if ! echo "$telemetry_trace" | curl -f -sSL --retry 5 -o /dev/null \
+    "$telemetry_url" \
+    --header 'Content-Type: application/json' \
+    --header "DD-Api-Key: $apikey" \
+    --data @-
+    then
+        echo "Unable to send trace telemetry\n"
+    fi
+}
+
+function configure_installer() {
+    local sudo_cmd="$1"
+    local trace_id="$2"
+    local apm_instrumentation_enabled="$3"
+
+    $sudo_cmd mkdir -p /etc/datadog-agent
+    $sudo_cmd bash -c "cat <<EOT > /etc/datadog-agent/datadog-installer.ini
+[features]
+apm_instrumentation=$apm_instrumentation_enabled
+
+[telemetry]
+trace_id=$trace_id
+parent_id=$trace_id
+priority=2
+EOT"
+}
+
+function _install_installer() {
+    local sudo_cmd="$1"
+    local os="$2"
+    local distribution="$3"
+    local apm_instrumentation_enabled="$4"
+
+    if [ "$os" == "Debian" ] && [ "$distribution" == "Debian" ] && [ -n "$apm_instrumentation_enabled" ]; then
+        echo -e "\033[34m\n* Installing the Datadog installer\n\033[0m"
+        configure_installer "$sudo_cmd" "$trace_id" "$apm_instrumentation_enabled"
+        $sudo_cmd apt-get install -o Acquire::Retries="5" -y --force-yes "datadog-installer"
+    fi
+}
+
+# Install the Datadog installer package
+function install_installer() {
+    local sudo_cmd="$1"
+    local os="$2"
+    local distribution="$3"
+    local apm_instrumentation_enabled="$4"
+
+    local trace_id
+    local start_time
+    local install_stdout
+    local install_stderr
+    local exit_status
+
+    trace_id=$(od -An -N8 -tu8 < /dev/urandom | tr -d ' ')
+    start_time=$(date +%s%N)
+
+    (_install_installer "$sudo_cmd" "$os" "$distribution" "$apm_instrumentation_enabled" > /tmp/datadog-installer-stdout.log 2> /tmp/datadog-installer-stderr.log) || exit_status=$?
+    exit_status=${exit_status:-0}
+
+    # escape newlines and double quotes before sending the telemetry
+    install_stdout=$(cat /tmp/datadog-installer-stdout.log | sed ':a;N;$!ba;s/\n/\\n/g' | sed 's/"/\\"/g')
+    install_stderr=$(cat /tmp/datadog-installer-stderr.log | sed ':a;N;$!ba;s/\n/\\n/g' | sed 's/"/\\"/g')
```

Comment:
This seems frail, maybe using `jq` to compose the telemetry payload would be more robust

Reply:
I initially discarded that option because we'd need to add it as a dependency to install and I'm not sure it's going to have a package available in all the distros we support. Do you think it's fine to add it as a dep ?